pipeline/src/pipeline.rs

```diff
@@ -322,8 +324,16 @@ impl<T: FieldElement> Pipeline<T> {
         self
     }
 
+    pub fn add_prover_data(&mut self, data: Vec<u8>) {
```

Comment:
I think we need to start adding documentation to the pipeline methods, most of them don't have it (in this PR, for the new and related methods)

Reply:
not sure about this name... its all prover data in the end right?

Reply:
done